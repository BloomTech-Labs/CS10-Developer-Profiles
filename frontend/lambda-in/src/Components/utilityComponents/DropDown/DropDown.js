import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Icon from '@material-ui/core/Icon';

/**
 * A wrapper component for @material-ui <ExpansionPanel />
 *
 * @description A Dropdown component wrapper, which accepts any type of children.
 *
 * @function A functional component.
 * @prop {string} header - Text to be displayed when the Dropdown is collapsed.
 *
 * @example How to use it
 * <DropDown header="Header text to display here">
 */
const DropDown = ({ header, children }) => (
  <Fragment>
    <div className="panelMargin">
      <ExpansionPanel>
        <ExpansionPanelSummary expandIcon={<Icon>add_circle</Icon>}>
          <Typography>{header}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <div className="expansionPanel">{children}</div>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </div>
  </Fragment>
);

DropDown.propTypes = {
  header: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.func,
  ]).isRequired,
};

export default DropDown;
