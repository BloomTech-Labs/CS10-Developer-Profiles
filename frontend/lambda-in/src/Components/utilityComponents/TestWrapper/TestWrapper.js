import React, { Component, Fragment } from 'react';

/**
 * A Functional stateless component wrapper
 *
 * @description Enzyme's 'shallow' method don't create an instance for
 * stateless functional componets
 * Whitout an instance is not possible to access the functional componet's
 * props.
 * Because enzyme's shallow generate an instance for class coponents, <TestWrapper/>
 * allow us to access any functional coponent passed as a children.
 * @see https://github.com/airbnb/enzyme/issues/1232#issuecomment-334510754
 */
class TestWrapper extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  /**
   * Catch children component error an log them in test-enviroment
   *
   * @description An Error bundary to catch potential children component's
   * Javascrip errors
   * @see https://reactjs.org/docs/error-boundaries.html
   * @param {object} error - Error ebject
   * @param {object} info - Component stack on error
   */
  componentDidCatch(error, info) {
    // eslint-disable-next-line no-console
    console.log('Error Bundary', error, info);
  }

  render() {
    // eslint-disable-next-line react/prop-types
    const { children } = this.props;
    return <Fragment>{children({ ...this.props })}</Fragment>;
  }
}

export default TestWrapper;
