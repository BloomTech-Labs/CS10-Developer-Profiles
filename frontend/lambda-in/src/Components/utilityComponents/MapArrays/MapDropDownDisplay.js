import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import DropDown from '../DropDown/DropDown';

const MapDropDownDisplay = (props) => {
  const { array, propsAndLabels } = props;

  // prettier-ignore
  return array.map(item => (
    <DropDown
      header={item.title || item.degree}
      key={`${item.title || item.degree}${Math.random()}`}
    >
      <div className="dropdown-item blur-handler">
        <Fragment>
          {// prettier-ignore
          propsAndLabels.map(propLabel => (
            <div
              className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
              key={`${Math.random()}-${propLabel[1]}`}
            >
              <div>
                <Typography variant="caption">{propLabel[1]}</Typography>
                <Typography variant="subheading">
                  {item[propLabel[0]]}
                </Typography>
              </div>
            </div>
          ))}
        </Fragment>
      </div>
    </DropDown>
  ));
};

MapDropDownDisplay.propTypes = {
  array: PropTypes.arrayOf(PropTypes.object).isRequired,
  propsAndLabels: PropTypes.arrayOf(PropTypes.array).isRequired,
};

export default MapDropDownDisplay;
