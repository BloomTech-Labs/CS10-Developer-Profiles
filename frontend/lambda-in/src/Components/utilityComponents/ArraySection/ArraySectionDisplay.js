import PropTypes from 'prop-types';
import React from 'react';
import SEEKER_PROP_TYPES from '../../seekerPropTypes';
import DropDown from '../DropDown/DropDown';
import MapDropDownDisplay from '../MapArrays/MapDropDownDisplay';

/**
 * Renders a user-array-section in individual blocks.
 * @param {onject} props - Required props
 */
const ArraySectionDisplay = (props) => {
  // prettier-ignore
  const {
    header,
    userInfo,
    field,
    schema,
  } = props;

  const propsAndLabels = Object.entries(schema);

  return (
    <DropDown header={header}>
      <MapDropDownDisplay
        array={userInfo[field]}
        propsAndLabels={propsAndLabels}
      />
    </DropDown>
  );
};

ArraySectionDisplay.propTypes = {
  header: PropTypes.string.isRequired,
  userInfo: SEEKER_PROP_TYPES.isRequired,
  field: PropTypes.string.isRequired,
  schema: PropTypes.shape({}).isRequired,
};

export default ArraySectionDisplay;
