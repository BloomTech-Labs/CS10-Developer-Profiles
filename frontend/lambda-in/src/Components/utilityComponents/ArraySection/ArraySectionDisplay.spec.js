import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import TestWrapper from '../TestWrapper/TestWrapper';
import ArraySectionDisplay from './ArraySectionDisplay';
import DropDown from '../DropDown/DropDown';
import MapDropDownDisplay from '../MapArrays/MapDropDownDisplay';

configure({ adapter: new Adapter() });

/**
 * @author jesuarva
 */
describe('ArraySectionDisplay component', () => {
  const userInfo = {
    projects: [
      {
        // tech: ['NODE', 'Express', 'Mongoose'],
        title: 'A nice project',
        description: 'A DRY library for express-mongoose',
        img: '/nice.img.edit',
        link: 'go.to.project.com',
        repo: 'github.my-repo',
      },
      {
        // tech: ['NODE', 'Express', 'Mongoose'],
        title: 'router Factroy',
        description: 'A DRY library for express-mongoose',
        img: '/nice.img',
        link: 'go.to.project.com',
        repo: 'github.my-repo',
      },
      {
        // tech: [],
        title: 'proj 1',
        description: 'desc 1',
        img: 'new.img',
        link: 'link',
        repo: '',
      },
      {
        // tech: [],
        title: 'today',
        description: 'today',
        img: 'today',
        link: 'today',
        repo: 'today',
      },
    ],
  };
  const schema = {
    title: 'Title',
    description: 'Description',
    img: 'Image',
    link: 'Link',
    repo: 'Repository',
    // tech: ['Stack'], // TODO
  };
  const props = {
    header: 'Projects',
    userInfo,
    field: 'projects',
    schema,
  };
  const arraySectionDisplay = (
    <ArraySectionDisplay
      header={props.header}
      userInfo={props.userInfo}
      field={props.field}
      schema={props.schema}
    />
  );

  const numberOfProjects = userInfo.projects.length;

  it('should renders correctly', () => {
    const tree = renderer.create(arraySectionDisplay).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Renders nested components', () => {
    const mounted = mount(
      <TestWrapper>{() => arraySectionDisplay}</TestWrapper>,
    );

    /**
     * For this specific test, there shall be 5 DropDown components rendered.
     * - 1 redered by ArraySectionDisplay.js (see line 22 on its file)
     * - 4 redered by MapDropDownDisplay.js (one for each project in 'userInfo.projects)
     */
    it('should renders <DropDown/> components', () => {
      const dropDownComponents = mounted.find(DropDown);
      expect(dropDownComponents).toHaveLength(1 + numberOfProjects);
    });

    it('should renders a <MapDropDownDisplay/> component', () => {
      const mapDropDownDisplayComponent = mounted.find(MapDropDownDisplay);
      expect(mapDropDownDisplayComponent).toHaveLength(1);
    });
  });
});
