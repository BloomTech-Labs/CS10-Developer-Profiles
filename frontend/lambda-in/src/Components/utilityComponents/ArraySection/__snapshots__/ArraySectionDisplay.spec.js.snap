// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArraySectionDisplay component should renders correctly 1`] = `
<div
  className="panelMargin"
>
  <div
    className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiExpansionPanelSummary-content-35"
      >
        <p
          className="MuiTypography-root-40 MuiTypography-body1-49"
        >
          Projects
        </p>
      </div>
      <div
        aria-hidden="true"
        className="MuiButtonBase-root-37 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-36"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label-71"
        >
          <span
            aria-hidden="true"
            className="material-icons MuiIcon-root-72"
          >
            add_circle
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-81"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      className="MuiCollapse-container-88"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper-90"
      >
        <div
          className="MuiCollapse-wrapperInner-91"
        >
          <div
            className="MuiExpansionPanelDetails-root-92"
          >
            <div
              className="expansionPanel"
            >
              <div
                className="panelMargin"
              >
                <div
                  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1"
                >
                  <div
                    aria-expanded={false}
                    className="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    <div
                      className="MuiExpansionPanelSummary-content-35"
                    >
                      <p
                        className="MuiTypography-root-40 MuiTypography-body1-49"
                      >
                        A nice project
                      </p>
                    </div>
                    <div
                      aria-hidden="true"
                      className="MuiButtonBase-root-37 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-36"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label-71"
                      >
                        <span
                          aria-hidden="true"
                          className="material-icons MuiIcon-root-72"
                        >
                          add_circle
                        </span>
                      </span>
                      <span
                        className="MuiTouchRipple-root-81"
                      />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    className="MuiCollapse-container-88"
                    style={
                      Object {
                        "minHeight": "0px",
                      }
                    }
                  >
                    <div
                      className="MuiCollapse-wrapper-90"
                    >
                      <div
                        className="MuiCollapse-wrapperInner-91"
                      >
                        <div
                          className="MuiExpansionPanelDetails-root-92"
                        >
                          <div
                            className="expansionPanel"
                          >
                            <div
                              className="dropdown-item blur-handler"
                            >
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Title
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    A nice project
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Description
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    A DRY library for express-mongoose
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Image
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    /nice.img.edit
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Link
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    go.to.project.com
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Repository
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    github.my-repo
                                  </h3>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="panelMargin"
              >
                <div
                  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1"
                >
                  <div
                    aria-expanded={false}
                    className="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    <div
                      className="MuiExpansionPanelSummary-content-35"
                    >
                      <p
                        className="MuiTypography-root-40 MuiTypography-body1-49"
                      >
                        router Factroy
                      </p>
                    </div>
                    <div
                      aria-hidden="true"
                      className="MuiButtonBase-root-37 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-36"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label-71"
                      >
                        <span
                          aria-hidden="true"
                          className="material-icons MuiIcon-root-72"
                        >
                          add_circle
                        </span>
                      </span>
                      <span
                        className="MuiTouchRipple-root-81"
                      />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    className="MuiCollapse-container-88"
                    style={
                      Object {
                        "minHeight": "0px",
                      }
                    }
                  >
                    <div
                      className="MuiCollapse-wrapper-90"
                    >
                      <div
                        className="MuiCollapse-wrapperInner-91"
                      >
                        <div
                          className="MuiExpansionPanelDetails-root-92"
                        >
                          <div
                            className="expansionPanel"
                          >
                            <div
                              className="dropdown-item blur-handler"
                            >
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Title
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    router Factroy
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Description
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    A DRY library for express-mongoose
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Image
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    /nice.img
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Link
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    go.to.project.com
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Repository
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    github.my-repo
                                  </h3>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="panelMargin"
              >
                <div
                  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1"
                >
                  <div
                    aria-expanded={false}
                    className="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    <div
                      className="MuiExpansionPanelSummary-content-35"
                    >
                      <p
                        className="MuiTypography-root-40 MuiTypography-body1-49"
                      >
                        proj 1
                      </p>
                    </div>
                    <div
                      aria-hidden="true"
                      className="MuiButtonBase-root-37 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-36"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label-71"
                      >
                        <span
                          aria-hidden="true"
                          className="material-icons MuiIcon-root-72"
                        >
                          add_circle
                        </span>
                      </span>
                      <span
                        className="MuiTouchRipple-root-81"
                      />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    className="MuiCollapse-container-88"
                    style={
                      Object {
                        "minHeight": "0px",
                      }
                    }
                  >
                    <div
                      className="MuiCollapse-wrapper-90"
                    >
                      <div
                        className="MuiCollapse-wrapperInner-91"
                      >
                        <div
                          className="MuiExpansionPanelDetails-root-92"
                        >
                          <div
                            className="expansionPanel"
                          >
                            <div
                              className="dropdown-item blur-handler"
                            >
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Title
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    proj 1
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Description
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    desc 1
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Image
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    new.img
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Link
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    link
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Repository
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    
                                  </h3>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="panelMargin"
              >
                <div
                  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiExpansionPanel-root-1"
                >
                  <div
                    aria-expanded={false}
                    className="MuiButtonBase-root-37 MuiExpansionPanelSummary-root-31"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    <div
                      className="MuiExpansionPanelSummary-content-35"
                    >
                      <p
                        className="MuiTypography-root-40 MuiTypography-body1-49"
                      >
                        today
                      </p>
                    </div>
                    <div
                      aria-hidden="true"
                      className="MuiButtonBase-root-37 MuiIconButton-root-66 MuiExpansionPanelSummary-expandIcon-36"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label-71"
                      >
                        <span
                          aria-hidden="true"
                          className="material-icons MuiIcon-root-72"
                        >
                          add_circle
                        </span>
                      </span>
                      <span
                        className="MuiTouchRipple-root-81"
                      />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    className="MuiCollapse-container-88"
                    style={
                      Object {
                        "minHeight": "0px",
                      }
                    }
                  >
                    <div
                      className="MuiCollapse-wrapper-90"
                    >
                      <div
                        className="MuiCollapse-wrapperInner-91"
                      >
                        <div
                          className="MuiExpansionPanelDetails-root-92"
                        >
                          <div
                            className="expansionPanel"
                          >
                            <div
                              className="dropdown-item blur-handler"
                            >
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Title
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    today
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Description
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    today
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Image
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    today
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Link
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    today
                                  </h3>
                                </div>
                              </div>
                              <div
                                className="mpadropdowndisplay-item--propperty inputFieldLargeMultiline"
                              >
                                <div>
                                  <span
                                    className="MuiTypography-root-40 MuiTypography-caption-50"
                                  >
                                    Repository
                                  </span>
                                  <h3
                                    className="MuiTypography-root-40 MuiTypography-subheading-47"
                                  >
                                    today
                                  </h3>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
